nano/node/bootstrap.cpp

```diff
@@ -2235,9 +2265,35 @@ class request_response_visitor : public nano::message_visitor
 	{
 	}
 	virtual ~request_response_visitor () = default;
-	void keepalive (nano::keepalive const &) override
+	void keepalive (nano::keepalive const & message_a) override
 	{
-		assert (false);
+		if (connection->node->config.logging.network_keepalive_logging ())
+		{
+			BOOST_LOG (connection->node->log) << boost::str (boost::format ("Received keepalive message from %1%") % connection->socket->remote_endpoint ());
+		}
+		connection->node->stats.inc (nano::stat::type::message, nano::stat::detail::keepalive, nano::stat::dir::in);
+		connection->node->network.merge_peers (message_a.peers);
+		nano::keepalive message;
+		connection->node->peers.random_fill (message.peers);
+		auto bytes = message.to_bytes ();
+		if (connection->node->config.logging.network_keepalive_logging ())
+		{
+			BOOST_LOG (connection->node->log) << boost::str (boost::format ("Keepalive req sent to %1%") % connection->socket->remote_endpoint ());
+		}
+		connection->socket->async_write (bytes, [connection = connection](boost::system::error_code const & ec, size_t size_a) {
```

Comment:
connection = connection (?)

Reply:
connection is a member variable, so if you want to pass it in without passing `this` this is the way to do it (and [&var = var] for reference)